src/Domains/Connectors/ScrapperApi/Workflows/Activities/ScrapperSearchActivity.php

```diff
@@ -30,7 +39,17 @@ public function execute(Model $model, AppInterface $app, array $params): array
                 $params['search']
             );
 
-            return $action->execute();
+            [$results] = Octane::concurrently([
+                fn () => $action->execute(),
+            ]);
+
+            $expiration = 3 * 24 * 60 * 60;
+
+            Redis::setex($word, $expiration, true);
```

Comment:
@FredPeal Instead of creating multiple individual keys, consider using a Redis hash/dictionary structure here. This would help avoid key sprawl and make the data organization more maintainable.